### DISCUSSION TITLE
Send data to handleChatMessage()

### DISCUSSION THREAD

**Discussion by romatexdev** [def-studio/telegraph#669] [Q&A] [answered]
How to send data with a message from a user in a chat and intercept it in handleChatMessage()?

I ask the user to enter a number
`Telegraph::message('Count')->send();`

Next, I intercept the text in handleChatMessage()
`protected function handleChatMessage(\Illuminate\Support\Stringable $text): void
    {
        $this->chat->html("Received: $text")->send();
    }`

Can you tell me how to send other data together and get it from handleChatMessage()?

When I receive in handleChatMessage(), the data contains only the entered text
`{
"update_id": 
564438053,
"message": 
{
"message_id": 
586,
"from": 
{
"id": 
9465464644,
"is_bot": 
false,
"first_name": 
"RDev",
"username": 
"romatexdev",
"language_code": 
"en"
},
"chat": 
{
"id": 
9465464644,
"first_name": 
"RDev",
"username": 
"romatexdev",
"type": 
"private"
},
"date": 
15546546464,
"text": 
"3"
}
}`

**Comment by fabio-ivona** [comment]
Hi, just reply form the handle chat message method 

$this->reply("hello");

And listen for the response inside handle chat message

For conversations you should better keep track of the conversation state, using telegraph storage or a custom implementation

**Comment by romatexdev** [comment]
And if I need to pass 2 or more parameters?

**Comment by romatexdev** [comment]
> Hi, just reply form the handle chat message method
> 
> $this->reply("hello");
> 
> And listen for the response inside handle chat message
> 
> For conversations you should better keep track of the conversation state, using telegraph storage or a custom implementation

Thank you for the answer!

But? i didn't quite understand how to use $this->reply("hello") in handleChatMessage()?
To clarify, I need to receive the payload along with the entered message from the user in handleChatMessage().

**Comment by fabio-ivona** [comment, marked as answer]
Reply does not use any payload of the received message
